### PR TITLE
MRG, ENH: Add resizing when time_viewer is present

### PR DESCRIPTION
**Pull request by larsoner**
On this PR this code snippet correctly sets the render area to the requested `size=(800, 600)`, and yet is resizeable afterward:

<details>
<summary>MWE</summary>

```
import os.path as op
import mne

data_path = mne.datasets.sample.data_path()
sample_dir = op.join(data_path, 'MEG', 'sample')
subjects_dir = op.join(data_path, 'subjects')
inv = mne.minimum_norm.read_inverse_operator(op.join(
    sample_dir, 'sample_audvis-meg-oct-6-meg-inv.fif'))
evoked = mne.read_evokeds(op.join(sample_dir, 'sample_audvis-ave.fif'))[0]
evoked.apply_baseline((None, 0))
stc = mne.minimum_norm.apply_inverse(
    evoked, inv, method='dSPM', verbose='debug', pick_ori='vector')  # 'vector')
initial_time = 0.1

size = (800, 600)
with mne.viz.use_3d_backend('pyvista'):
    brain = stc.plot(subjects_dir=subjects_dir, initial_time=initial_time,
                     clim='auto', views='lat', hemi='lh',
                     smoothing_steps='nearest', verbose=True,
                     size=size)
sc = brain.screenshot()
assert sc.shape[:2] == size[::-1]
```

</details>

Screenshot from 2020-05-12 14-17-05

@GuillaumeFavelier can you take over to actually incorporate this properly? For example this also needs to work when `time_viewer=False` (it does not currently) -- you can probably use similar techniques. But I'll let you decide how to refactor to make it work for both cases.

### CODE REVIEW

File: mne/viz/_brain/_timeviewer.py [hunk 1]
```diff
@@ -355,7 +357,21 @@ def __init__(self, brain, show_traces=False):
...
+        sz = self.brain._size
+        adjust_mpl = self.show_traces and not self.separate_canvas
+        if adjust_mpl:
+            self.mpl_canvas.canvas.setMinimumSize(
+                sz[0], int(round(sz[1] / self.interactor_stretch)))
+        try:
+            yield
+        finally:
+            if adjust_mpl:
+                self.mpl_canvas.canvas.setMinimumSize(0, 0)
```
Comment (larsoner) [inline mne/viz/_brain/_timeviewer.py:374]: This flow control seems cleaner to me, WDYT?
```suggestion
        if not adjust_mpl:
            yield
        else:
            self.mpl_canvas.canvas.setMinimumSize(
                sz[0], int(round(sz[1] / self.interactor_stretch)))
            try:
                yield
            finally:
                self.mpl_canvas.canvas.setMinimumSize(0, 0)
```

Comment (GuillaumeFavelier) [reply mne/viz/_brain/_timeviewer.py:374]: One if to rule them all

**Comment by larsoner** [comment]
@GuillaumeFavelier can you add tests for size? I started to add some but the size was wrong for one run, things don't work with latest pyvista + pyvistaqt, etc. so it was not trivial

**Comment by GuillaumeFavelier** [comment]
I'll take care of it

**Pull request merged**